### ISSUE TITLE
Battery sensor shows unexpected behavior

### ISSUE THREAD

**Issue by kauthmbt** [bropat/eufy-security-ws#57]
**Describe the bug**  
After updating from 0.3.3 to the latest version the battery sensor has problems showing the correct state.

**To Reproduce**  
1. Updating to latest addon + integration (by fuatakgun)

**Expected behavior**  
Battery sensor showing the correct percentage.

**Screenshots & Logfiles**  

![eufy1](https://user-images.githubusercontent.com/74350030/143940190-135af99c-b0b8-4379-a688-7ea8eee42947.jpg)

```
2021-11-29 20:23:45.609  INFO Eufy Security server listening on port 3000 
2021-11-29 20:23:47.173  INFO Connected to station T8010 on host 192.x.x.x and port 11294 
2021-11-29 20:23:49.047  INFO Push notification connection successfully established 

```

**Versions:**  
 - Client version: 0.6.0 (Eufy Security WS Add-on by fuatakgun)
 - Operating system: Home Assistant OS 6.6 & core-2021.11.5

**Hardware information about camera**
 - T8210

**Additional context**  
I already tried a fallback and reinstalled the old addon(s) from a backup (both Eufy security WS addon + integration). After a reboot the battery sensor will immediately show the correct percentage again:

![eufy2](https://user-images.githubusercontent.com/74350030/143940528-ab87434d-6b17-4127-910a-99c58fc3f17b.jpg)

After some usage time it seems like periodically the correct value is shown and after a while drops to 0% again:
![eufy3](https://user-images.githubusercontent.com/74350030/143940682-b11009c3-9dc9-4b28-8549-befe5e6390aa.png)
![eufy4](https://user-images.githubusercontent.com/74350030/143941002-6585dd87-01ae-434b-bc7b-d82ead1bb5df.jpg)

**Comment by fuatakgun** [comment]
I am having the same problem, incoming events are captured below, i will try to capture debug logs.

```
2021-11-29 17:19:27 - set_event_for_entity - device / T8113N63205014E2 / battery / 52
2021-11-30 09:57:36 - set_event_for_entity - device / T8113N63205014E2 / battery / 51
2021-11-30 21:46:51 - set_event_for_entity - device / T8113N63205014E2 / battery / 0
2021-11-30 21:56:05 - set_event_for_entity - device / T8113N63205014E2 / battery / 51
2021-11-30 22:11:59 - set_event_for_entity - device / T8113N63205014E2 / battery / 0
2021-11-30 23:32:49 - set_event_for_entity - device / T8113N63205014E2 / battery / 67
2021-12-01 08:10:40 - set_event_for_entity - device / T8113N63205014E2 / battery / 100

```

**Comment by kauthmbt** [comment]
Additionally I found out that when the ringing sensor is set to TRUE the battery sensor is (almost) immediately updated with the most current battery percentage.
![ringing_sensor](https://user-images.githubusercontent.com/74350030/144419625-d2bad8aa-f03b-4f74-8137-094f079839f8.jpg)

![battery01](https://user-images.githubusercontent.com/74350030/144419653-82d092e5-f31c-4f87-a55e-ae24b9317e3a.jpg)
![battery02](https://user-images.githubusercontent.com/74350030/144419656-29db5fea-c530-4634-a272-779a5dcfafe7.jpg)

**Comment by bropat** [comment]
@kauthmbt 

The battery value is received via 2 interfaces in the `eufy-security-client` driver:

* Cloud
* P2P connection

It looks like one of the 2 interfaces is now delivering an incorrect value.

To understand exactly which one returns the wrong value I need the debug logs where the problem is visible (the log extract should contain the correct value plus the 0 value of the battery property).

To activate the debug mode, the Docker container must have the parameter `DEBUG` set with some value (e.g.: "y"). As already described [here](https://github.com/bropat/eufy-security-ws#docker).

**Comment by kauthmbt** [comment]
@bropat 
My challenge over here is that the protocol log window from the addon only shows a limited number of log entries. Maybe @fuatakgun  can show me an alternative way if the following information is not yet sufficient for further analysis.

```
2021-12-08 08:29:15.603  DEBUG [P2PClientProtocol.handleMsg] Station T8010P************ - LOOKUP_ADDR - Got invalid ip address 0.0.0.0, ignoring response... 
2021-12-08 08:29:26.059  DEBUG [PushClient.handleParsedMessage] Login response: GCM -> logged in -> waiting for push messages... 
2021-12-08 08:29:26.063  DEBUG [HTTPApi.request] Request: 
{
  method: 'post',
  endpoint: 'v1/apppush/register_push_token',
  baseUrl: 'https://security-app-eu.eufylife.com',
  token: '** REMOVED **',
  data: {
    is_notification_enable: true,
    token: '** REMOVED **',
    transaction: '1638952166061'
  },
  headers: {
    app_version: 'v3.3.1_1058',
    os_type: 'android',
    os_version: '30',
    phone_model: 'EUFYCLIENT',
    country: 'DE',
    language: 'en',
    openudid: 'c854eb256e52cbd7',
    uid: '',
    net_type: 'wifi',
    mnc: '02',
    mcc: '262',
    sn: '** REMOVED **',
    Model_type: 'PHONE',
    timezone: 'GMT+01:00',
    'Cache-Control': 'no-cache',
    'User-Agent': 'okhttp/3.12.1',
    'X-Auth-Token': '** REMOVED **'
  }
}
2021-12-08 08:29:26.098  DEBUG [PushClient.handleParsedMessage] IqStanza: Not implemented 
{
  tag: 7,
  object: {
    type: 'SET',
    id: '',
    extension: {
      id: 12,
      data: <Buffer >
    }
  }
}
2021-12-08 08:29:26.133  DEBUG [HTTPApi.registerPushToken] Response: 
{
  code: 0,
  msg: 'Succeed.'
}
2021-12-08 08:29:26.136  DEBUG [HTTPApi.registerPushToken] Push token registered successfully 
2021-12-08 08:29:26.139  DEBUG [HTTPApi.request] Request: 
{
  method: 'post',
  endpoint: 'v1/app/review/app_push_check',
  baseUrl: 'https://security-app-eu.eufylife.com',
  token: '** REMOVED **',
  data: {
    app_type: 'eufySecurity',
    transaction: '1638952166138'
  },
  headers: {
    app_version: 'v3.3.1_1058',
    os_type: 'android',
    os_version: '30',
    phone_model: 'EUFYCLIENT',
    country: 'DE',
    language: 'en',
    openudid: 'c854eb256e52cbd7',
    uid: '',
    net_type: 'wifi',
    mnc: '02',
    mcc: '262',
    sn: '** REMOVED **',
    Model_type: 'PHONE',
    timezone: 'GMT+01:00',
    'Cache-Control': 'no-cache',
    'User-Agent': 'okhttp/3.12.1',
    'X-Auth-Token': '** REMOVED **'
  }
}
2021-12-08 08:29:26.175  DEBUG [HTTPApi.checkPushToken] Response: 
{
  code: 0,
  msg: 'Succeed.'
}
2021-12-08 08:29:26.178  DEBUG [HTTPApi.checkPushToken] Push token OK 
2021-12-08 08:29:26.180  INFO [PushNotificationService.<anonymous>] Push notification connection successfully established 
2021-12-08 08:34:26.076  DEBUG [PushClient.sendHeartbeat] Sending heartbeat... 2 
2021-12-08 08:34:26.083  DEBUG [PushClient.buildHeartbeatPingRequest] heartbeatPingRequest 
{
  last_stream_id_received: 2
}
2021-12-08 08:34:26.139  DEBUG [PushClient.handleHeartbeatAck] Heartbeat acknowledge 
{
  tag: 1,
  object: {
    lastStreamIdReceived: 2
  }
}

[MISSING PARTS]

      timestamp: 1612626833000
    },
    chargingStatus: {
      value: 3,
      timestamp: 1638189458000
    },
    ringtoneVolume: {
      value: 0,
      timestamp: 1620291321000
    },
    videoWdr: {
      value: true,
      timestamp: 1619260446000
    },
    statusLed: {
      value: true,
      timestamp: 1612465348000
    },
    battery: {
      value: 89,
      timestamp: 1638952155191
    },
    watermark: {
      value: 1,
      timestamp: 1612509575000
    },
    chimeHomebaseRingtoneType: {
      value: 0,
      timestamp: 1635590889000
    },
    state: {
      value: 1,
      timestamp: 1612465342000
    },
    audioRecording: {
      value: true,
      timestamp: 1612465348000
    },
    autoNightvision: {
      value: true,
      timestamp: 1612465348000
    },
    speakerVolume: {
      value: 100,
      timestamp: 1612465348000
    },
    videoStreamingQuality: {
      value: 2,
      timestamp: 1628006878000
    },
    powerWorkingMode: {
      value: 1,
      timestamp: 1612465348000
    },
    batteryTemperature: {
      value: 6,
      timestamp: 1638156751000
    },
    wifiRssi: {
      value: -48,
      timestamp: 1638952155191
    },
    wifiSignalLevel: {
      value: 4,
      timestamp: 1638185700000
    },
    motionDetectionType: {
      value: 0,
      timestamp: 1612465342000
    },
    motionDetected: {
      value: false,
      timestamp: 0
    },
    personDetected: {
      value: false,
      timestamp: 0
    },
    personName: {
      value: '',
      timestamp: 0
    },
    ringing: {
      value: false,
      timestamp: 0
    }
  }
}
2021-12-08 08:39:26.091  DEBUG [PushClient.sendHeartbeat] Sending heartbeat... 3 
2021-12-08 08:39:26.095  DEBUG [PushClient.buildHeartbeatPingRequest] heartbeatPingRequest 
{
  last_stream_id_received: 3
}
2021-12-08 08:39:26.135  DEBUG [PushClient.handleHeartbeatAck] Heartbeat acknowledge 
{
  tag: 1,
  object: {
    lastStreamIdReceived: 3
  }
}
2021-12-08 08:44:26.101  DEBUG [PushClient.sendHeartbeat] Sending heartbeat... 4 
2021-12-08 08:44:26.105  DEBUG [PushClient.buildHeartbeatPingRequest] heartbeatPingRequest 
{
  last_stream_id_received: 4
}
2021-12-08 08:44:26.146  DEBUG [PushClient.handleHeartbeatAck] Heartbeat acknowledge 
{
  tag: 1,
  object: {
    lastStreamIdReceived: 4
  }
}

[MISSING PARTS]

      timestamp: 1628006780000
    },
    chimeIndoor: {
      value: false,
      timestamp: 1612465348000
    },
    motionDetectionSensitivity: {
      value: 3,
      timestamp: 1612465348000
    },
    chimeHomebaseRingtoneVolume: {
      value: 14,
      timestamp: 1612626833000
    },
    chargingStatus: {
      value: 3,
      timestamp: 1638189458000
    },
    ringtoneVolume: {
      value: 0,
      timestamp: 1620291321000
    },
    videoWdr: {
      value: true,
      timestamp: 1619260446000
    },
    statusLed: {
      value: true,
      timestamp: 1612465348000
    },
    battery: {
      value: 89,
      timestamp: 1638952155191
    },
    watermark: {
      value: 1,
      timestamp: 1612509575000
    },
    chimeHomebaseRingtoneType: {
      value: 0,
      timestamp: 1635590889000
    },
    state: {
      value: 1,
      timestamp: 1612465342000
    },
    audioRecording: {
      value: true,
      timestamp: 1612465348000
    },
    autoNightvision: {
      value: true,
      timestamp: 1612465348000
    },
    speakerVolume: {
      value: 100,
      timestamp: 1612465348000
    },
    videoStreamingQuality: {
      value: 2,
      timestamp: 1628006878000
    },
    powerWorkingMode: {
      value: 1,
      timestamp: 1612465348000
    },
    batteryTemperature: {
      value: 6,
      timestamp: 1638156751000
    },
    wifiRssi: {
      value: -48,
      timestamp: 1638952155191
    },
    wifiSignalLevel: {
      value: 4,
      timestamp: 1638185700000
    },
    motionDetectionType: {
      value: 0,
      timestamp: 1612465342000
    },
    motionDetected: {
      value: false,
      timestamp: 0
    },
    personDetected: {
      value: false,
      timestamp: 0
    },
    personName: {
      value: '',
      timestamp: 0
    },
    ringing: {
      value: false,
      timestamp: 0
    }
  }
}
```

Looks for me that the battery value shown as "89" is indeed correctly registered. However the entity _sensor.doorbell_battery_ still shows "0"

**Comment by bropat** [comment]
@kauthmbt 

I need the entire log of `eufy-security-ws`. What you posted above is not enough.

**Comment by fuatakgun** [comment]
I was able to pin down my specific issue, when my 2c camera is being charged using non-default charger, it was jumping between 0 and correct value and charge was very slow. I have used direct usb charger and I can see that it is not happening. @kauthmbt , is your camera being charged while this issue is happening?

![image](https://user-images.githubusercontent.com/11085566/145866868-7a542541-6c35-4269-9d8e-33bbf6d1ec9c.png)

**Comment by skrobul** [comment]
fwiw, upgrading the custom component (fuatakgun/eufy_security) to latest version(`ff37f6`) solved the "0 battery" problem for me. There were some updates to it since this issue has been opened. My troubleshooting workflow connecting directly to `eufy-security-ws` with [client](https://github.com/bropat/eufy-security-ws#start-client) and checking that battery is reported correctly which means the problem is likely on the home assistant side

**Comment by kauthmbt** [comment]
@fuatakgun 
The video doorbell is wired to the doorbell.

Nevertheless: The default behavior is no permanent charging of the battery. Normally the charging process only starts around 70% residual power. Then it charges up to 90% and switches the charging process off again.

@skrobul
Thanks for the hint - I will give it a try and report about the outcome.

**Comment by fuatakgun** [comment]
Yes, my initial assessment was wrong, issue repeated again, i will share debug logs soon.

**Comment by fuatakgun** [comment]
> fwiw, upgrading the custom component (fuatakgun/eufy_security) to latest version(`ff37f6`) solved the "0 battery" problem for me. There were some updates to it since this issue has been opened. My troubleshooting workflow connecting directly to `eufy-security-ws` with [client](https://github.com/bropat/eufy-security-ws#start-client) and checking that battery is reported correctly which means the problem is likely on the home assistant side

This is not a valid validation method to find out root cause. Home assistant integration doesn't impose any business logic other than exposing sensors and applying push notifications. Somewhere earlier, this push notification was received as zero battery but checking over eufy-security-ws client cannot have the same problem as current state is different.

**Comment by fuatakgun** [comment]
I have been waiting this thing to happen for whole day, it did not happen :) of course

**Comment by bropat** [comment]
Perhaps this problem is only present with doorbells?

**Comment by fuatakgun** [comment]
Nope, having same problem with my 2c camera, not recently though. I am following this closely and try to come up with debug logs

**Comment by eikowagenknecht** [comment]
Any news on this? It also affects some of my devices:
- T8114 - Some showing correct percentage, some not
- T8140-Z - Some showing correct percentage, some not

**Comment by fuatakgun** [comment]
It is on me to generate debug logs and a bit busy nowadays.

**Comment by fuatakgun** [comment]
Email is sent

**Comment by trullock** [comment]
I also have this issue. How do I get https://github.com/fuatakgun/eufy_security_addon into debug mode so I can report something useful?
Thanks

**Comment by fuatakgun** [comment]
The problem is that, when you enable debug mode inside home assistant add-on, you will not be able to get all the log lines. I am skipping this to add intentionally as it would not add any value to add debug option.

You should run this as a docker instance to generate valuable log files otherwise even though i enable debug option in add-on, you will not be able to capture all of them.

**Comment by trullock** [comment]
so just `docker run -it -e USERNAME=email@address.com -e PASSWORD=password_goes_here -p 3000:3000 bropat/eufy-security-ws:0.7.0`? 

Will that spurt sufficient logs?

Should I disable my HA addon first, do they interfere with each other?

**Comment by fuatakgun** [comment]
if you are running in the same host, you have to disable the add-on, yes.

For debug logs, use this below;

```
docker run -it -e DEBUG=1 -e USERNAME=email@address.com -e PASSWORD=password_goes_here -p 3000:3000 bropat/eufy-security-ws:0.7.0
```

**Comment by trullock** [comment]
Brill, logs obtained and emailed

**Comment by bropat** [comment]
Issue found will be fixed in next release (https://github.com/bropat/eufy-security-client/issues/109).

**Comment by bropat** [comment]
Version 0.8.0 released

**Comment by kauthmbt** [comment]
First of all: sorry for re-opening this issue again.
If it is not intended let me know and I will create an additional one and link it.

It looked good after the release of 0.8.0. At least I got no dropping to "0" since the update (06.02.2022).
At least yesterday (with 0.8.2) and today (0.8.3) the sensor dropped to "0" again in case of the doorbell pressed event. Can't really be sure about the same behaviour was already valid for 0.8.0 because this was the first time doorbell was pressed after the update.

<img width="832" alt="image" src="https://user-images.githubusercontent.com/74350030/153183660-58b5e632-03a6-4ad7-bcaf-07647d8eb8d9.png">

As I have seen that additional users encountered the same behaviour with T8210 it might be interesting if they can reproduce the same issue on their devices as well.
@eikowagenknecht , @trullock , @TuomasPakkanen ?

**Comment by trullock** [comment]
I can confirm on 0.8.2 I'm getting a value for battery

**Comment by kauthmbt** [comment]
@trullock 
Does it even keep the value when the doorbell is pressed?

**Comment by trullock** [comment]
I assume so, as its been pressed since installing 0.8.2 and it still has a value

**Comment by TuomasPakkanen** [comment]
Hi,

Thanks for pinging me as well. Thought also that this was already fixed but looking at the doorbell battery sensor now it has dropped to zero from about 25 percent this morning. The doorbell itself was not pressed then but it did probably register a motion event when I left home.

**Comment by bropat** [comment]
The doorbell sends some power events via P2P. It could be that something was misinterpreted here and it therefore comes to these 0 values.
So that I can refine it further and then finally solve the problem, I need from you the debug logs where I see exactly which P2P event with which data has caused it.
I don't have a doorbell myself and therefore can't check it.

**Comment by kauthmbt** [comment]
I am trying to fetch the logs with the instructions @fuatakgun provided above (after adjusting the ws version of course) at least for my scenario.
Hoping that I will ab able to manage on my own because I am not a command-line pro...

**Comment by eikowagenknecht** [comment]
I'm on 0.8.3 and the battery states are still completely unreliable (showing 0% where the camery has almost full battery):
![image](https://user-images.githubusercontent.com/1475672/153872087-988315ff-1f51-4aca-83bd-28971e66e6ab.png)

**Comment by kauthmbt** [comment]
Debug log finally obtained and emailed.

**Comment by Beer17HWAM** [comment]
@bropat Did you find the problem?

**Comment by emufan** [comment]
Same here. 2 out of 4 2C cameras dropped to 0. The other two still show values like 81%, but I don't know, if this is only better than the faulty 2 or correct.

**Comment by KnalleArie** [comment]
When will the next release be available with the fix?

**Comment by 5746787689** [comment]
Sorry for commenting on a closed issue, and if required I am more than happy to raise a new case. 

However I have recently set up a T8210 doorbell and I appear to have the same issue as described above, where the battery level does not update and remains static... currently mine shows as 60%, even though I have recently charged it fully.

I am running v2.8.0 of https://github.com/fuatakgun/eufy_security

Is there anything I can do to provide useful debug information @fuatakgun ?

Thanks for your help!
